# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/sessionstore/SessionStoreDataCollector.cpp
# Commit: b3de4da02449
# Full Hash: b3de4da024493e23b6ff4f3649cb083a74bfaee9
# Author: Kashav Madan <kshvmdn@gmail.com>
# Date: 2021-06-22 21:29:07
# Description:
#   Bug 1717497 - Only assert we have a timer if we expect to have created one, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D118482
# ==============================================================================

diff -r 63e5df9ef639 -r b3de4da02449 toolkit/components/sessionstore/SessionStoreDataCollector.cpp
--- a/toolkit/components/sessionstore/SessionStoreDataCollector.cpp	Tue Jun 22 14:56:44 2021 +0000
+++ b/toolkit/components/sessionstore/SessionStoreDataCollector.cpp	Tue Jun 22 15:08:41 2021 +0000
@@ -48,11 +48,15 @@
   uint32_t epoch =
       aWindowChild->BrowsingContext()->Top()->GetSessionStoreEpoch();
   if (listener) {
-    MOZ_DIAGNOSTIC_ASSERT(listener->mTimer);
+    MOZ_DIAGNOSTIC_ASSERT_IF(
+        !StaticPrefs::browser_sessionstore_debug_no_auto_updates(),
+        listener->mTimer);
     if (listener->mEpoch == epoch) {
       return listener.forget();
     }
-    listener->mTimer->Cancel();
+    if (listener->mTimer) {
+      listener->mTimer->Cancel();
+    }
   }
 
   listener = new SessionStoreDataCollector(aWindowChild, epoch);
